kedro/framework/cli/utils.py

```diff
@@ -431,32 +431,16 @@ def _reformat_load_versions(ctx, param, value) -> dict[str, str]:
 def _split_params(ctx, param, value):
     if isinstance(value, dict):
         return value
-    dot_list = []
-    for item in split_string(ctx, param, value):
-        equals_idx = item.find("=")
-        colon_idx = item.find(":")
-        if equals_idx != -1 and colon_idx != -1 and equals_idx < colon_idx:
-            # For cases where key-value pair is separated by = and the value contains a colon
-            # which should not be replaced by =
-            pass
-        else:
-            item = item.replace(":", "=", 1)  # noqa: PLW2901
-        items = item.split("=", 1)
-        if len(items) != 2:  # noqa: PLR2004
-            ctx.fail(
-                f"Invalid format of `{param.name}` option: "
-                f"Item `{items[0]}` must contain "
-                f"a key and a value separated by `:` or `=`."
-            )
-        key = items[0].strip()
-        if not key:
-            ctx.fail(
-                f"Invalid format of `{param.name}` option: Parameter key "
-                f"cannot be an empty string."
-            )
```

Comment:
I'm not sure `omegaconf` does equally strict error handling and I would expect us to still provide the error messages above when a user doesn't provide a proper parameter key value pair.